[GStreamer] imported/w3c/web-platform-tests/mediacapture-record/MediaRecorder-blob-timecode.https.html is a permanent failure

philn opened the pull request:
#### 237403719d8f655504bd4f4526a3f25523d96a8a
<pre>
[GStreamer] imported/w3c/web-platform-tests/mediacapture-record/MediaRecorder-blob-timecode.https.html is a permanent failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=282911">https://bugs.webkit.org/show_bug.cgi?id=282911</a>

Reviewed by Xabier Rodriguez-Calvar.

Ensure the first blob event has a time code set to 0. Following ones use the position reported by
the transcoder.

* LayoutTests/platform/glib/TestExpectations:
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.cpp:
(WebCore::MediaRecorderPrivateBackend::stopRecording):
(WebCore::MediaRecorderPrivateBackend::fetchData):
(WebCore::MediaRecorderPrivateBackend::notifyPosition):
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.h:
(WebCore::MediaRecorderPrivateBackend::WTF_GUARDED_BY_LOCK):
(WebCore::MediaRecorderPrivateBackend::notifyPosition): Deleted.

Canonical link: <a href="https://commits.webkit.org/286587@main">https://commits.webkit.org/286587@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d4d7fb2158f603f5368f8bc88ea4c700114e0556

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/76436 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/55471 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29342 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80961 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27712 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/78553 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/64613 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3764 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/59931 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18046 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/79503 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/49844 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65637 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40260 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/47239 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/23126 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/26035 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/68365 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23458 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/82409 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3812 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/2499 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/68213 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3965 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65606 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67456 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16822 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11424 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/9523 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3759 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/6568 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3782 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/7212 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/5540 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cadubentzen commented on Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.cpp:
```diff
@@ -210,7 +210,9 @@ void MediaRecorderPrivateBackend::fetchData(MediaRecorderPrivate::FetchDataCallb
-        completionHandler(WTFMove(buffer), mimeType, m_position);
+        completionHandler(WTFMove(buffer), mimeType, m_timeCode);
+        if (m_position.isValid())
+            m_timeCode = m_position.toDouble();
```
If the position has been updated before the lock is acquired, shouldn't we update the timecode before calling the completion handler? i.e. having the if condition here before the completion handler call.

Edit: I get it now, you want to force timecode 0 for the first.

cadubentzen replied:
Also, I'm wondering whether the data lock being held while the completion handler is called could be causing the deadlocks/timeouts if it reenters this function on the handler. Maybe the lock can be released before calling it?

philn replied:
Oh yes you're right, that's why the test is Pass/Timeout after this patch. Updating as you suggested makes it consistent Pass here :)

cadubentzen reviewed (approved):
_unofficial r+ (not a reviewer)_

philn committed:
[GStreamer] imported/w3c/web-platform-tests/mediacapture-record/MediaRecorder-blob-timecode.https.html is a permanent failure
https://bugs.webkit.org/show_bug.cgi?id=282911

Reviewed by Xabier Rodriguez-Calvar.

Ensure the first blob event has a time code set to 0. Following ones use the position reported by
the transcoder.

* LayoutTests/platform/glib/TestExpectations:
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.cpp:
(WebCore::MediaRecorderPrivateBackend::stopRecording):
(WebCore::MediaRecorderPrivateBackend::fetchData):
(WebCore::MediaRecorderPrivateBackend::notifyPosition):
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateGStreamer.h:
(WebCore::MediaRecorderPrivateBackend::WTF_GUARDED_BY_LOCK):
(WebCore::MediaRecorderPrivateBackend::notifyPosition): Deleted.

Canonical link: https://commits.webkit.org/286587@main
